### PR TITLE
basis_universal: Clarify encoder-only dependencies, only used in editor builds

### PR DESCRIPTION
**Pull request by akien-mga**
Proper solution to the issue #99825 tried to address. CC @gio3k

### CODE REVIEW

File: modules/basis_universal/SCsub
Comment (akien-mga) [inline modules/basis_universal/SCsub:60]: Note: This will change behavior in dev builds. `BASISU_DEVEL_MESSAGES` doesn't exist in the current version, so what this was meant to be is `BASISU_FORCE_DEVEL_MESSAGES`.

I'm not sure we actually want to see those in dev builds though, so alternatively we might remove this define to keep the current behavior.

File: modules/basis_universal/SCsub
Comment (akien-mga) [inline modules/basis_universal/SCsub:42]: Let's see if `jpeg-compressor` really needed the `-isystem` trick or if we can get away with how I refactored this.

File: modules/basis_universal/SCsub
Comment (akien-mga) [reply modules/basis_universal/SCsub:59]: I'm erring on the side of removing both those flags actually. I don't think most engine developers benefit from a verbose basisu debugging output. Anyone working _on_ the basisu integration can enable those temporarily.

**Review by clayjohn — APPROVED** [review]
The changes make sense to me, and if it builds I think we can be confident that it works!

**Review by Repiteo — APPROVED** [review]
Refactor shenanigans seemed to do the trick. Gonna wait for confirmation whether you wanna get rid of those flags before merging

File: modules/basis_universal/SCsub
Comment (akien-mga) [reply modules/basis_universal/SCsub]: Yeah let's give it a try, and reconsider if someone finds they're missing important dev-only info.

**Pull request merged**
